File: Trees/LC_104.cpp
// Written by: Jian-Yuan Lin
// Date: 2022.11.29

class Solution {
public:
    int maxDepth(TreeNode* root) {
        return cal_dep(root);
    }
private:
    int cal_dep(TreeNode* root){
        if (root == nullptr){
            return 0;
        }
        int l = cal_dep(root->left), r = cal_dep(root->right);
        return 1 + max(l, r);
    }
};

File: Trees/LC_104.py
# Written by: Jian-Yuan Lin
# Date: 2022.11.29

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        def find_depth(root):
            if not root:
                return 0
            l , r = find_depth(root.left), find_depth(root.right)
            return 1 + max(l, r)
        return find_depth(root)